crates/matrix-sdk-ui/tests/integration/timeline/mod.rs

```diff
@@ -307,6 +308,76 @@ async fn test_redact_message() {
     assert_matches!(timeline_stream.next().await, Some(VectorDiff::Remove { index: 2 }));
 }
 
+#[async_test]
+async fn test_redact_local_sent_message() {
+    let room_id = room_id!("!a98sd12bjh:example.org");
+    let (client, server) = logged_in_client_with_server().await;
+    let sync_settings = SyncSettings::new().timeout(Duration::from_millis(3000));
+
+    let mut sync_builder = SyncResponseBuilder::new();
+    sync_builder.add_joined_room(JoinedRoomBuilder::new(room_id));
+
+    mock_sync(&server, sync_builder.build_json_sync_response(), None).await;
+    let _response = client.sync_once(sync_settings.clone()).await.unwrap();
+    server.reset().await;
+
+    mock_encryption_state(&server, false).await;
+
+    let room = client.get_room(room_id).unwrap();
+    let timeline = room.timeline().await.unwrap();
+    let (_, mut timeline_stream) = timeline.subscribe().await;
+
+    // Mock event sending.
+    Mock::given(method("PUT"))
+        .and(path_regex(r"^/_matrix/client/r0/rooms/.*/send/.*"))
+        .and(header("authorization", "Bearer 1234"))
+        .respond_with(
+            ResponseTemplate::new(200)
+                .set_body_json(json!({ "event_id": "$wWgymRfo7ri1uQx0NXO40vLJ" })),
+        )
+        .expect(1)
+        .mount(&server)
+        .await;
+
+    // Send the event so it's added to the send queue as a local event.
+    timeline
+        .send(RoomMessageEventContent::text_plain("i will disappear soon").into())
+        .await
+        .unwrap();
+
+    // Assert the local event is in the timeline now and is not sent yet.
+    assert_let_timeout!(Some(VectorDiff::PushBack { value: item }) = timeline_stream.next());
+    let event = item.as_event().unwrap();
+    assert!(event.is_local_echo());
+    assert_matches!(event.send_state(), Some(EventSendState::NotSentYet));
+
+    // As well as a day divider.
+    assert_let_timeout!(
+        Some(VectorDiff::PushFront { value: day_divider }) = timeline_stream.next()
+    );
+    assert!(day_divider.is_day_divider());
+
+    // Let the send queue send the event so it's now a local event with a remote
+    // handle (event_id).
+    yield_now();
+
+    // We receive an update in the timeline from the send queue.
+    assert_let_timeout!(Some(VectorDiff::Set { index, value: item }) = timeline_stream.next());
```

Comment:
Do we need the yield_now now that we're using `assert_let_timeout`?

Reply:
Oh, it seems like we don't. I'm removing them, thanks!